### ISSUE TITLE
Names - search, sort

### ISSUE THREAD

**Issue by cytadela8** [aeternity/mdw-frontend#61]
https://explorer.aeternity.io/names

Currently you can only see a list of names in simingly random order. It would be nice to be able to search and sort by different properties.

**Comment by marc0olo** [comment]
people interested in this can meanwhile use https://aenalytics.org to search for names and see everything related to

**Comment by ifaouibadi** [comment]
added this ability to sort by name ASC & DESC and the names that will expire soon
![A3DA3865-D7CA-47A5-9477-538CB5E05514](https://user-images.githubusercontent.com/10424827/149754474-40a4a003-9c62-42a6-a828-64a23539fe6b.png)

we cannot do search/filter by name right now as it's not supported by the API yet!

**Comment by marc0olo** [comment]
> we cannot do search/filter by name right now as it's not supported by the API yet!

actually there is a basic search for names that could be used to filter. it works with the starting chars:

- https://mainnet.aeternity.io/mdw/names/search/crypto

be aware that names can have following states:

- active
- in auction
- expired

we could first start with displaying only the active names here.

**Comment by marc0olo** [comment]
do you want to add this in the PR or should I review and merge the PR independently? @ifaouibadi

**Comment by ifaouibadi** [comment]
will be adding it shortly

**Comment by marc0olo** [comment]
@ifaouibadi there is a difference in how the response looks like

it is all returned in the same API, look.

**Active name in search:
- https://mainnet.aeternity.io/mdw/names/search/infinitecoin

```json
[{"active":true,"auction":null,"hash":"nm_2dog79SQCtmrR5EUYiTZmfPSCWuuWSQYG8tp4GHRQMp8QTHMZ5","info":{"active_from":545070,"auction_timeout":null,"claims":[29503883],"expire_height":725080,"ownership":{"current":"ak_21y7Rs7wHH5DWAgM84KXQQBbKCwZo759j4G4DZ6NEBvZz5QLoZ","original":"ak_21y7Rs7wHH5DWAgM84KXQQBbKCwZo759j4G4DZ6NEBvZz5QLoZ"},"pointers":{"account_pubkey":"ak_21y7Rs7wHH5DWAgM84KXQQBbKCwZo759j4G4DZ6NEBvZz5QLoZ"},"revoke":null,"transfers":[],"updates":[29529335]},"name":"infinitecoin.chain","previous":[],"status":"name"}]
```

**Auction in search**
- https://mainnet.aeternity.io/mdw/names/search/cr7

```json
[{"active":false,"info":{"auction_end":546221,"bids":[28034755],"last_bid":{"block_hash":"mh_2Php3RFdFJKANNLV8K8rM2zSjWcxMhCar3WhVhwZMQTB5XLJgL","block_height":516461,"hash":"th_2H37GC9A9bc5NfY7ERhTyqr9qqKhLtrR57cFpf9NuzeGY9Pbef","micro_index":2,"micro_time":1637047217038,"signatures":["sg_Vgkgo77xNBKRf4qjNZCN6ijACmmwV2MyfRsY2n5quzEdw48MV9eHH1Tq8HV6Co6zAc9dPvypCrdiGfvKhDZVWGAGsRoD2"],"tx":{"account_id":"ak_sffz9ERe6ArQZduvtZBs75N5NGFpbCKRf3PaUC6btGs6X4YLC","fee":16500000000000,"name":"cr7.chain","name_fee":217830900000000000000,"name_id":"nm_di1QQwjJy5p5nZwwehKZAJeTYR5ZLJmzRn5rzkD5vxYzoHgCw","name_salt":3355629924087400,"nonce":5,"ttl":726221,"type":"NameClaimTx","version":2},"tx_index":28034755}},"name":"cr7.chain","previous":[],"status":"auction"},{"active":false,"auction":null,"hash":"nm_9oxueZQDFbtU91FxKLDsBcAAJwHvnkhCNmbSxWv4MtC14ddvo","info":{"active_from":20088,"auction_timeout":0,"claims":[278331],"expire_height":132275,"ownership":{"current":"ak_2Qn9ouHZ2D6bh2CqQ8vX9A4Ybar1Qy1pztAkowFv3X7LVTQarm","original":"ak_2Qn9ouHZ2D6bh2CqQ8vX9A4Ybar1Qy1pztAkowFv3X7LVTQarm"},"pointers":{"account_pubkey":"ak_2Qn9ouHZ2D6bh2CqQ8vX9A4Ybar1Qy1pztAkowFv3X7LVTQarm"},"revoke":null,"transfers":[],"updates":[1844639,852135,680002,312313]},"name":"cr7.test","previous":[],"status":"name"}]
```

**Expired name in search**
- https://mainnet.aeternity.io/mdw/names/search/cryptodrugs

```json
[{"active":false,"auction":null,"hash":"nm_ApisZAf6WTGuR8otm1VWwP5a993npZYk1hkiajnJaZJcRyhC3","info":{"active_from":165050,"auction_timeout":null,"claims":[4927276],"expire_height":215050,"ownership":{"current":"ak_uZ3E6Q9MVov8R7nxaq8CgdEVqWwJ85aiNRWCdRqFPHkNkENdv","original":"ak_uZ3E6Q9MVov8R7nxaq8CgdEVqWwJ85aiNRWCdRqFPHkNkENdv"},"pointers":{},"revoke":null,"transfers":[],"updates":[]},"name":"cryptodrugs.chain","previous":[],"status":"name"}]
```

**Comment by ifaouibadi** [comment]
@marc0olo this PR ready for merge

**Comment by marc0olo** [comment]
@ifaouibadi same here, let's discuss solutions in the corresponding PR in the future. I am really struggling right now to switch between PR and issue to see what's going on 😅

**Comment by marc0olo** [comment]
closed with #145
